### PR TITLE
chore(service_naming): multiple contribs

### PR DESCRIPTION
**Pull request by tabgok**
chore(service naming): Multiple Contribs

== Special Note ==
These changes are gated behind a feature flag and are not expected to be
stable until a release note is generated.  Existing customers should be
unaffected.

== Service Naming Summary ==
Datadog is updating its method of determining service names!

Service naming introduces a few concepts:

* Versioned service name "schemas":
  * v0: The existing mechanism for determining service names
  * v1: Replacing the 'default' contrib-specific service name with the env var $DD_SERVICE

These changes are being in conjunction with changes to peer.service updates to bring a better service-naming experience to customers. In particular, these changes will help:

* Remove the creation of fake services, such as grpc.client
* Standardize service names across and within tracers
* Remove the need to create as many fake services to take advantage of Datadog features, such as metrics
* Make more sense to humans

== Operation Naming Summary ==
The service name changes mean that generated spans can no longer rely on `span.service` to differentiate between processes.  A subclass of spans now have a new problem - differentiating between inbound and outbound requests (for the purposes of metrics).  For instance, spans with names `http.request` used to be differentiated by the service they belonged to (incoming requests under `span.service=$DD_SERVICE` and outgoing under `span.service=$DD_CONTRIB_SERVICE_NAME`).  These are now no longer differentiated.

To remedy these issues, operation names are also being changed as part of the service naming effort.  In the above example, `http.request` turns into `http.server.request` (incoming) and `http.client.request` (outgoing).

== This Change ==
This change updates the following contribs:
* urllib3
* requests
* httpx
* httplib
* wsgi
* tornado
* sanic
* pyramid
* pylons
* molten
* flask
* falcon
* djangorestframework
* django
* cherrypy
* bottle
* asgi
* aiohttpvertica
## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] OPTIONAL: PR description includes explicit acknowledgement of the performance implications of the change as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: tests/subprocesstest.py [hunk 1]
```diff
@@ -94,19 +97,22 @@ def _full_method_name(self):
...
         # variable for the subprocess to detect.
         env_overrides = self._get_env_overrides()
+        use_pytest = self._get_use_pytest()
+        full_testcase_name = self._full_method_name(use_pytest=use_pytest)
         sp_test_env = os.environ.copy()
         sp_test_env.update(env_overrides)
         sp_test_env[SUBPROC_ENV_VAR] = "True"
-        sp_test_cmd = ["python", "-m", "unittest", full_testcase_name]
+        test_framework = "pytest" if use_pytest else "unittest"
+        sp_test_cmd = ["python", "-m", test_framework, full_testcase_name]
```
Comment (tabgok) [inline tests/subprocesstest.py:115]: I needed this change to be able to use fixtures in the subprocess test.

I tried running all subprocess tests with pytests, but several (~5) contrib test suites failed.

File: ddtrace/contrib/vertica/patch.py [hunk 2]
```diff
@@ -75,32 +77,36 @@ def cursor_span_end(instance, cursor, _, conf, *args, **kwargs):
...
-                        "operation_name": "vertica.query",
+                        "operation_name": schematize_database_operation("vertica.query", database_provider="vertica"),
                         "span_type": SpanTypes.SQL,
                         "span_start": execute_span_start,
                         "span_end": execute_span_end,
                         "measured": True,
                     },
                     "copy": {
-                        "operation_name": "vertica.copy",
+                        "operation_name": schematize_database_operation("vertica.copy", database_provider="vertica"),
```
Comment (ZStriker19) [inline ddtrace/contrib/vertica/patch.py:87]: Noting `copy` and `nextset` spans will now add into the `vertica.query` metrics. Does this seem right?

Comment (ZStriker19) [reply ddtrace/contrib/vertica/patch.py:87]: I wonder if "query" is the right name.... Does it count as querying if you're writing to a database? Feel free to ignore, just thinking.

Comment (tabgok) [reply ddtrace/contrib/vertica/patch.py:87]: No, this is a good catch, I'm taking copy out

File: tests/contrib/aiohttp/test_middleware.py [hunk 3]
```diff
@@ -44,6 +46,62 @@ async def test_handler(app_tracer, aiohttp_client):
...
+def test_service_operation_schema(ddtrace_run_python_code_in_subprocess, schema_version):
+    """
+    vO/None:
+        operation: module_name.request
+        service name: DD_SERVICE
+    v1:
+        operation: http.server.request
+        service name: DD_SERVICE
+    """
+    expected_service_name = {
```
[the remark this answers is not in the document]
Comment (tabgok) [reply tests/contrib/aiohttp/test_middleware.py:59]: =p

File: tests/contrib/cherrypy/test_middleware.py [hunk 4]
```diff
@@ -467,12 +469,59 @@ def test_service_configuration_middleware(self):
 
         cherrypy.tools.tracer.service = previous_service
 
+    # @TracerTestCase.run_in_subprocess(env_overrides=dict(DD_SERVICE='mysvc'))
```
[the remark this answers is not in the document]
Comment (tabgok) [reply tests/contrib/cherrypy/test_middleware.py:472]: yes =/

File: tests/contrib/flask/test_request.py [hunk 5]
```diff
@@ -1073,3 +1075,94 @@ def hello():
...
+@pytest.mark.parametrize("service_name", [None, "mysvc"])
+@pytest.mark.parametrize("schema_version", [None, "v0", "v1"])
+def test_schematized_service_name(ddtrace_run_python_code_in_subprocess, schema_version, service_name):
+    """
+    v0/Default: expect the service name to be "flask"
+    v1: expect the service name to be "unnamed-python-service"
+    """
+
+    expected_service_name = {
+        None: service_name or "flask",
```
[the remark this answers is not in the document]
Comment (tabgok) [reply tests/contrib/flask/test_request.py:1089]: service_name is a parameter that is passed in for the purposes of the test, at test runtime it ends up being `None` or a string

File: tests/contrib/pyramid/test_pyramid.py [hunk 6]
```diff
@@ -119,6 +121,74 @@ def test_distributed_tracing_disabled(self):
...
+class TestSchematization(PyramidBase):
+    instrument = True
+
+    @TracerTestCase.run_in_subprocess(env_overrides=dict(DD_SERVICE="mysvc"))
+    def test_schematized_service_name_default(self):
+        self.app.get("/", status=200)
+        spans = self.pop_spans()
+        assert len(spans) == 1
+        s = spans[0]
+        assert s.service == "pyramid", s.service
```
[the remark this answers is not in the document]
Comment (tabgok) [reply tests/contrib/pyramid/test_pyramid.py:133]: We're asserting `s.service == "pyramid"`.  If that fails, then the assert will print out `s.service` (it helps with debugging).  I can make the debug clearer via `"Expected 'pyramid' and got {}".format(s.service)`

File: tests/contrib/tornado/test_tornado_web.py [hunk 7]
```diff
@@ -659,3 +660,89 @@ def test_custom_default_handler(self):
...
+
+    @TracerTestCase.run_in_subprocess(env_overrides=dict(DD_SERVICE="mysvc"))
+    def test_service_name_schematization_default(self):
+        self.fetch("/success/")
+        traces = self.pop_traces()
+        assert 1 == len(traces)
+        assert 1 == len(traces[0])
+
+        request_span = traces[0][0]
+        assert "mysvc" == request_span.service, request_span.service
```
[the remark this answers is not in the document]
Comment (tabgok) [reply tests/contrib/tornado/test_tornado_web.py:678]: same as aboe, changing these to:
`"Expected 'mysvc' and got {}".format(s.service)`

File: tests/contrib/pyramid/test_pyramid.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (ZStriker19) [reply tests/contrib/pyramid/test_pyramid.py]: ah sorry, I forgot about this syntax, all good

**Review by emmettbutler — COMMENTED** [review]
Looks like there are some merge conflicts. I'm going to hold off on reviewing this until they're resolved.

**Review by ZStriker19 — APPROVED** [review]

**Pull request merged**